### DISCUSSION TITLE
Displaying sprites

### DISCUSSION THREAD

**Discussion by Besdroxk** [mapeditor/tiled#3826] [Q&A]
I am new to qt framework. I was very interested in this project and tried to make my own simplified analog (I have a goal to create my own engine for 2d games in the future, so I think my analog will be part of it). I was able to recreate the basic functionality, but I noticed a very big problem - with a large map (even without sprites, with sprites it's even worse)) I have very large lags). To store sprites and display them, I use QTableView + QAbstractTableModel (implemented through a vector). And since I'm a beginner, it's hard for me to navigate the qt code. Can you explain to me how exactly the display and saving of sprites is implemented in your project. (sorry for my English :))

**Comment by bjorn** [comment]
Since you closed this you might have already found the cause of your performance issues? There are so many reasons why performance can be bad, that it's hard to provide a generic answer to this.

For example, it can be important to make incremental updates when adding/removing or changing sprites, to avoid doing lots of needless updating, but incremental updates can in themselves also become a performance issue, which can then often be solved by batching things together.

In case the issue was just from rendering tiles, it is important to render only what's needed rather than the entire map. In Tiled, I use the "exposed rect" information when rendering tile layers to greatly reduce the painting time.
